enterprise/api_client/discovery.py

```diff
@@ -121,6 +121,29 @@ def get_search_results(self, querystring=None, traverse_pagination=True):
             many=False,
         )
 
+    def get_catalog_results(self, query):
+        """
+        Return results from the discovery service's search/all endpoint.
+
+        Arguments:
+            query (dict): query parameters used to filter search results.
+
+        Returns:
+            dict: The paginated response.
+        """
+        response = {}
+
+        try:
+            endpoint = getattr(self.client, self.SEARCH_ALL_ENDPOINT)
+            response = endpoint().post(query)
+        except Exception as ex:  # pylint: disable=broad-except
```

Comment:
Instead of a broad exception lets short list it, e.g.`except (SlumberBaseException, ConnectionError, Timeout) as ex`

Reply:
We are already doing the same thing in edx-platform [here](https://github.com/edx/edx-platform/blob/master/openedx/core/lib/edx_api_utils.py#L74)